src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -531,6 +545,68 @@ export function updateGeneQuery(
     return updatedQueries;
 }
 
+// This function acts as a toggle. If present in 'geneQueries', the query
+// is removed. If absent, a gene1/gene2 query is added.
+export function updateStructuralVariantQuery(
+    geneQueries: SingleGeneQuery[],
+    selectedGene1: string | undefined,
+    selectedGene2: string | undefined
+): SingleGeneQuery[] {
+    if (!selectedGene1 && !selectedGene2) {
+        return geneQueries;
+    }
+
+    // TODO replace with STRUCTVARAnyGeneStr (does not work somehow...)
+    const gene1 = selectedGene1 || '*';
+    const gene2 = selectedGene2 || '*';
+
+    // Remove any SV alteration with the same genes
+    // (both upstream and downstream fusions are evaluated).
+    const updatedQueries = _.filter(geneQueries, (query: SingleGeneQuery) => {
+        const isStructVar = queryContainsStructVarAlteration(query);
+        const isUpstreamMatch =
+            query.gene === gene1 &&
+            query.alterations &&
+            !!_.find(
```

Comment:
i think ```_.some``` is nicer than ```!!_.find```

Reply:
Yes, better. I replaced this and any other occurence in the code base.